django/views/generic/edit.py

```diff
@@ -162,7 +163,15 @@ def get_success_url(self):
         Returns the supplied URL.
         """
         if self.success_url:
-            url = self.success_url % self.object.__dict__
+            if re.search(r'%\([^\)]+\)', self.success_url):
```

Comment:
You don't check for the conversion flag and type specifiers. But I think that's fine. Would get a bit noisy otherwise and the current approach seems unlikely to give false positives.